monai/handlers/roc_auc.py

```diff
@@ -65,54 +61,16 @@ def __init__(
         output_transform: Callable = lambda x: x,
         device: Optional[torch.device] = None,
     ) -> None:
-        super().__init__(output_transform, device=device)
-        self.to_onehot_y = to_onehot_y
-        self.softmax = softmax
-        self.other_act = other_act
-        self.average: Average = Average(average)
-
-    @reinit__is_reduced
-    def reset(self) -> None:
-        self._predictions: List[torch.Tensor] = []
-        self._targets: List[torch.Tensor] = []
-
-    @reinit__is_reduced
-    def update(self, output: Sequence[torch.Tensor]) -> None:
-        """
-        Args:
-            output: sequence with contents [y_pred, y].
-
-        Raises:
-            ValueError: When ``output`` length is not 2. ROCAUC metric can only support y_pred and y.
-            ValueError: When ``y_pred`` dimension is not one of [1, 2].
-            ValueError: When ``y`` dimension is not one of [1, 2].
-
-        """
-        if len(output) != 2:
-            raise ValueError(f"output must have length 2, got {len(output)}.")
-        y_pred, y = output
-        if y_pred.ndimension() not in (1, 2):
-            raise ValueError("Predictions should be of shape (batch_size, n_classes) or (batch_size, ).")
-        if y.ndimension() not in (1, 2):
-            raise ValueError("Targets should be of shape (batch_size, n_classes) or (batch_size, ).")
-
-        self._predictions.append(y_pred.clone())
-        self._targets.append(y.clone())
-
-    def compute(self):
-        _prediction_tensor = torch.cat(self._predictions, dim=0)
-        _target_tensor = torch.cat(self._targets, dim=0)
-
-        if dist.is_available() and dist.is_initialized() and not self._is_reduced:
-            _prediction_tensor = all_gather(_prediction_tensor)
-            _target_tensor = all_gather(_target_tensor)
-            self._is_reduced = True
-
-        return compute_roc_auc(
-            y_pred=_prediction_tensor,
-            y=_target_tensor,
-            to_onehot_y=self.to_onehot_y,
-            softmax=self.softmax,
-            other_act=self.other_act,
-            average=self.average,
+        super().__init__(
+            compute_fn=lambda pred, label: compute_roc_auc(
```

Comment:
I feel for readability this should be defined as a local function rather than a lambda. Above this line I would put 

```py
def _compute_fc(pred, label):
    return compute_roc_auc(...)
```

then the argument `compute_fn` refers to this.

Reply:
Sure, updated in the latest commit.
Thanks.